src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1241,4 +1241,26 @@ public static void onEntityEnterSection(Entity entity, long packedOldPos, long p
         MinecraftForge.EVENT_BUS.post(new EntityEvent.EnteringSection(entity, packedOldPos, packedNewPos));
     }
 
+    public static String encodeLifecycle(Lifecycle lifecycle)
+    {
+        if(lifecycle == Lifecycle.stable())
+            return "stable";
+        if(lifecycle == Lifecycle.experimental())
+            return "experimental";
+        if(lifecycle instanceof Lifecycle.Deprecated dep)
+            return "deprecated=" + dep.since();
+        throw new IllegalArgumentException("Unknown lifecycle.");
+    }
+
+    public static Lifecycle parseLifecycle(String lifecycle)
+    {
+        if(lifecycle.equals("stable"))
+            return Lifecycle.stable();
+        if(lifecycle.equals("experimental"))
+            return Lifecycle.experimental();
+        if(lifecycle.startsWith("deprecated="))
+            return Lifecycle.deprecated(Integer.parseInt(lifecycle.substring(lifecycle.indexOf('=')+1)));
```

Comment:
Formatting: space before and after `+` operator:
```suggestion
            return Lifecycle.deprecated(Integer.parseInt(lifecycle.substring(lifecycle.indexOf('=') + 1)));
```